src/pages/permissions/panels/CreateTLSIdentityPanel.tsx

```diff
@@ -0,0 +1,162 @@
+import { useNotify } from "@canonical/react-components";
+import SidePanel from "components/SidePanel";
+import type { FC } from "react";
+import usePanelParams from "util/usePanelParams";
+import * as Yup from "yup";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import type { TLSIdentityFormValues } from "../forms/IdpGroupForm";
+import IdpGroupForm from "../forms/IdpGroupForm";
+import GroupSelection from "./GroupSelection";
+import useEditHistory from "util/useEditHistory";
+import GroupSelectionActions from "../actions/GroupSelectionActions";
+import { useGroups } from "context/useGroups";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { base64EncodeObject } from "util/helpers";
+
+interface GroupEditHistory {
+  groupsAdded: Set<string>;
+}
+
+interface Props {
+  openPortal: () => void;
+  setToken: React.Dispatch<React.SetStateAction<string>>;
+  setError: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const CreateTLSIdentityPanel: FC<Props> = ({
+  openPortal,
+  setToken,
+  setError,
+}) => {
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+
+  const { data: groups = [], error, isLoading } = useGroups();
+
+  const {
+    desiredState,
+    save: saveToPanelHistory,
+    undo: undoMappingChanges,
+  } = useEditHistory<GroupEditHistory>({
+    initialState: {
+      groupsAdded: new Set(),
+    },
+  });
+
+  if (error) {
+    notify.failure("Loading panel details failed", error);
+  }
+
+  const modifyGroups = (newGroups: string[], isUnselectAll?: boolean) => {
+    if (isUnselectAll) {
+      saveToPanelHistory({
+        groupsAdded: new Set(),
+      });
+    } else {
+      saveToPanelHistory({
+        groupsAdded: new Set(newGroups),
+      });
+    }
+  };
+
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const handleSubmit = (values: TLSIdentityFormValues) => {
+    const newIdentity = {
+      name: values.name,
+      groups: Array.from(desiredState.groupsAdded),
+    };
+    setToken("");
+
+    closePanel();
+    setError(false);
+    createFineGrainedTlsIdentity(newIdentity)
+      .then((response) => {
+        const encodedToken = base64EncodeObject(response);
+        setToken(encodedToken);
+
+        queryClient.invalidateQueries({
+          queryKey: [queryKeys.identities],
+        });
+      })
+      .catch(() => {
+        setError(true);
+      });
+
+    openPortal();
+  };
+
+  const groupSchema = Yup.object().shape({
+    name: Yup.string().required("Identity name is required"),
+  });
+
+  const formik = useFormik<TLSIdentityFormValues>({
+    initialValues: {
+      name: "",
```

Comment:
```suggestion
      name: "",
      groups: [],
```
I think we should init the groups as well.